app/tests/participants_tests/test_forms.py

```diff
@@ -0,0 +1,173 @@
+import pytest
+
+from grandchallenge.participants.forms import RegistrationRequestForm
+from grandchallenge.participants.models import (
+    RegistrationQuestionAnswer,
+    RegistrationRequest,
+)
+from tests.factories import (
+    ChallengeFactory,
+    RegistrationQuestionFactory,
+    UserFactory,
+)
+
+
+def answers_formset_data(*args, n=None):
+    if n is None:
+        n = len(args)
+    return {
+        "registration_question_answers-TOTAL_FORMS": str(n),
+        "registration_question_answers-INITIAL_FORMS": "0",
+        "registration_question_answers-MIN_NUM_FORMS": str(n),
+        "registration_question_answers-MAX_NUM_FORMS": str(n),
+        **{
+            f"registration_question_answers-{i}-answer": a
+            for i, a in enumerate(args)
+        },
+    }
```

Comment:
This is a bit too smart for a test. Easier for future programmers to work with explicit data.

Reply:
I think we are in different camps with this. I'm more of the: single place to change the interface to the form. But I'll bend to the common denom in the code base.

Reply:
Done!

Reply:
I would agree if this were application code, but test code is different. It is much easier to debug if you can see exactly what is going on. If you have some centralised utility in a test, if you modify that then you need to check all of the tests where it is used in case you change any of the assumptions about what the test is testing in the first place. Same reason that a test should have no-to-little logic: setup, mutate, assert.

Reply:
> If you have some centralised utility in a test, if you modify that then you need to check all of the tests where it is used in case you change any of the assumptions about what the test is testing in the first place.

That is a good argument. A lot of 'out of scope' tests might still work when you change them, but you might have accidentally broken a test somewhere. Had not thought about that danger. Thanks!

I'll try and grog it a bit more next time!